runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpUtils.kt

```diff
@@ -162,6 +166,23 @@ internal class OkHttpProxySelector(
     override fun connectFailed(uri: URI?, sa: SocketAddress?, ioe: IOException?) {}
 }
 
+internal class OkHttpCall(
+    request: HttpRequest,
+    response: HttpResponse,
+    requestTime: Instant,
+    responseTime: Instant,
+    coroutineContext: CoroutineContext = EmptyCoroutineContext,
+    val call: Call,
+) : HttpCall(request, response, requestTime, responseTime, coroutineContext) {
+    override fun copy(request: HttpRequest, response: HttpResponse): HttpCall =
+        OkHttpCall(request, response, requestTime, responseTime, coroutineContext, call)
+    override fun cancelInFlight() {
+        super.cancelInFlight()
+        coroutineContext.logger<OkHttpCall>().trace { "cancelling in-flight call" }
```

Comment:
**Suggestion**: Seems like this trace message would be better emitted in the superclass.